pkg/provider/aws/resources/eks.go

```diff
@@ -628,24 +627,25 @@ func (cluster *EksCluster) InstallAlbController(references core.AnnotationKeySet
 	}
 
 	role, err := core.CreateResource[*IamRole](dag, RoleCreateParams{
-		AppName: cluster.Name,
+		AppName: appName,
 		Name:    "alb-controller",
 		Refs:    references,
 	})
 	if err != nil {
 		return nil, err
 	}
-	assumeRolePolicyDoc := GetServiceAccountAssumeRolePolicy(saName, cluster.getOidc(dag))
-	if err != nil {
-		return nil, err
-	}
+	oidc, err := core.CreateResource[*OpenIdConnectProvider](dag, OidcCreateParams{
+		AppName:     appName,
+		ClusterName: strings.TrimLeft(cluster.Name, fmt.Sprintf("%s-", appName)),
+		Refs:        role.ConstructsRef.Clone(),
+	})
 	var aRef core.AnnotationKey
 	for ref := range references {
 		aRef = ref
 		break
 	}
+	dag.AddDependency(role, oidc)
```

Comment:
I'm confused why the role would need to depend on these - doesn't seem like it references it.

Reply:
the roles assume role policy can reference it when its tied to a service account. Realistically we will want to pass in the service account name as edge data as well, we likely need to figure out how we want to structure edge data because that will address one of the comments above

Reply:
I'd expect that dependency to be added when the reference is added - where does that happen and is there a reason the dependency couldn't be added there?

Reply:
this is where that all happens. when we install the ALB controller we know we need to create a service account and a role. In order to tie the 2 together we have to have the role depend on the oidc

Reply:
this will be able to be cleaned up long term when we dont only work in terms of helm charts and manifests in our KB

Reply:
Ah, the connection happens in `[*resources.IamRole, *resources.OpenIdConnectProvider].Configure` where it sets the policy doc. It's a bit odd to me that they're so separate - hard to track down.